server/planning/content_profiles/planning_types_async_service.py

```diff
@@ -0,0 +1,132 @@
+    Entries can be overridden by providing alternates in the planning_types mongo collection.
+    """
+
+    async def find_one(
+        self,
+        req: SearchRequest | None = None,
+        projection: ProjectedFieldArg | None = None,
+        use_mongo: bool = False,
+        version: int | None = None,
+        **lookup,
+    ) -> PlanningTypesResourceModel | None:
+        """
+        Overrides the `find_one` method to merge default planning type configurations
+        with database entries. If no entry exists in the database, it returns a default
+        planning type configuration.
+        """
+        try:
+            search_request = (
+                req
+                if req is not None
+                else SearchRequest(
+                    where=lookup,
+                    page=1,
+                    max_results=1,
+                    projection=projection,
+                    use_mongo=use_mongo,
+                    version=version,
+                )
+            )
+
+            planning_type = await super().find_one(search_request)
+
+            # lookup name from either **lookup of planning_item(if lookup has only '_id')
+            lookup_name = lookup.get("name")
+            if not lookup_name and planning_type:
+                lookup_name = planning_type.to_dict().get("name")
+
+            default_planning_type = deepcopy(
+                next(
+                    (ptype for ptype in DEFAULT_PROFILES if ptype.get("name") == lookup_name),
+                    {},
+                )
+            )
+            if not planning_type:
+                await self._remove_unsupported_fields(default_planning_type)
+                return PlanningTypesResourceModel(**default_planning_type)
+
+            await self.merge_planning_type(planning_type.to_dict(), default_planning_type)
+            return planning_type
+        except IndexError:
+            return None
+
+    async def get(self, req, lookup) -> ListCursor:
+        cursor = await super().search(lookup)
+        planning_types = await cursor.to_list_raw()
+        merged_planning_types = []
+
+        for default_planning_type in deepcopy(DEFAULT_PROFILES):
+            planning_type = next(
+                (p for p in planning_types if p.get("name") == default_planning_type.get("name")),
+                None,
+            )
+
+            # If nothing is defined in database for this planning_type, use default
+            if planning_type is None:
+                await self._remove_unsupported_fields(default_planning_type)
+                merged_planning_types.append(default_planning_type)
+            else:
+                await self.merge_planning_type(planning_type, default_planning_type)
+                merged_planning_types.append(planning_type)
+
+        return ListCursor(merged_planning_types)
+
+    async def merge_planning_type(self, planning_type: dict[str, Any], default_planning_type: dict[str, Any]):
```

Comment:
These next 2 methods do not wait on any I/O, therefor these don't need to be async